Correct liquidation price in the UI

da-kami opened the pull request:
Fixes https://github.com/itchysats/itchysats/issues/1641

The liquidation price is only relevant for the UI:
- Remove the liquidation price from the `Order` (finally...)
- Send the liquidation price according to own position in the projection (the only thing that makes sense)

da-kami committed:
Correct liquidation price in the UI

The liquidation price is only relevant for the UI:
- Remove the liquidation price from the `Order` (finally...)
- Send the liquidation price according to own position in the projection (the only thing that makes sense)

da-kami commented on daemon-tests/tests/happy_path.rs:
```diff
@@ -169,6 +169,9 @@ fn assert_eq_orders(mut published: CfdOrder, received: CfdOrder) {
     published.initial_funding_fee_per_lot = received.initial_funding_fee_per_lot;
 
+    // align liquidation price so we can assert on the order
+    published.liquidation_price = received.liquidation_price;
```
Note: I thought about if we can easily assert on the actual values, but it cannot easily be achieved and I don't think the actor tests are responsible for that so I did not invest time.

luckysori commented on model/src/cfd.rs:
```diff
@@ -266,19 +263,13 @@ impl Order {
             max_quantity,
             leverage_taker: leverage_choices_for_taker,
             trading_pair: TradingPair::BtcUsd,
-            liquidation_price,
```
❓ Is this a breaking change for our API or database?

da-kami replied:
good point, the `Order` is still part of `MakerOffers` on the wire, but since that API changed anyway I think it should be fine to change this. 
My understanding is, that we will have two makers running and old takers are going against the old maker, new takers against the new maker. That should be fine. The automation is fine because the projection API did not change.

Or am I missing something @klochowicz  @thomaseizinger ? 🤔


For the db it does not matter. We are not saving the `Order` anymore.

thomaseizinger replied:
Yep, this is all correct AFAIK. No relevant breaking changes here!

da-kami committed:
Decision `margin_per_lot` based on lon/short

The logic for deciding the maker's leverage should be found to one place, which is the `long_and_short_leverage` that decides the long and short leverage based on the role.
Once we know the long and short leverage that should be made for decision-making.
